ssl/ssl_lib.cc

```diff
@@ -3315,3 +3315,34 @@ int SSL_CTX_set1_curves_list(SSL_CTX *ctx, const char *curves) {
 int SSL_set1_curves_list(SSL *ssl, const char *curves) {
   return SSL_set1_groups_list(ssl, curves);
 }
+
+size_t SSL_client_hello_get0_ciphers(SSL *ssl, const unsigned char **out) {
+  STACK_OF(SSL_CIPHER) *client_cipher_suites = SSL_get_client_ciphers(ssl);
+  if (!client_cipher_suites) {
+      return 0;
+  }
+
+  if (out) {
+    // Called before
+    if(!ssl->client_cipher_suites_arr.empty()) {
+      *out = reinterpret_cast<unsigned char*>(ssl->client_cipher_suites_arr.data());
+      return ssl->client_cipher_suites_arr.size();
+    }
+
+    size_t total_length = sk_SSL_CIPHER_num(client_cipher_suites);
+    assert(ssl->client_cipher_suites_arr.Init(total_length));
```

Comment:
Is assert the right check here? Should this always be a failure?

Reply:
If out is not null (so caller is requesting writing the ciphers to out), array initialization would only fail if there is a memory allocation error. So if this fails, error should be raised? We could also choose to return 0?

Reply:
Yeah if memory allocation fails this function should gracefully handle it and return 0